gensim/downloader.py

```diff
@@ -164,6 +164,36 @@ def _calculate_md5_checksum(fname):
     return hash_md5.hexdigest()
 
 
+def _load_info(url=DATA_LIST_URL, encoding='utf-8'):
+    """Load dataset information from the network.
+
+    If the network access fails, fall back to a local cache.  This cache gets
+    updated each time a network request _succeeds_.
+    """
+    cache_path = os.path.join(base_dir, 'information.json')
+
+    try:
+        info_bytes = urlopen(DATA_LIST_URL).read()
+    except (OSError, IOError) as err:
+        #
+        # The exception raised by urlopen differs between Py2 and Py3.
+        #
+        # https://docs.python.org/3/library/urllib.error.html
+        # https://docs.python.org/2/library/urllib.html
+        #
+        logger.error('caught non-fatal exception, see trace below')
+        logger.exception(err)
+        logger.error('attempting to recover from local cache (%r)', cache_path)
+    else:
+        with open(cache_path, 'wb') as fout:
+            fout.write(info_bytes)
+
+        return json.loads(info_bytes.decode(encoding))
```

Comment:
Not needed: cleaner to fall through to the main case below, after caching locally.